addons/LPCAnimatedSprite/LPCAnimatedSprite2D.gd

```diff
@@ -55,51 +69,146 @@ func _notification(what):
 func _enter_tree():
 	if Engine.is_editor_hint():
 		LoadAnimations()
-	
+
 func LoadAnimations():
 	AnimationNames = LPCAnimation.keys()
 	var children = get_children();
 	for child in children:
 		remove_child(child)
 		
-	for spriteSheet in SpriteSheets:
-		if spriteSheet == null:
+	if 0 == SpriteSheets.size():
+		push_error("No detected LPCSpriteSheets.")
+	else:
+		if SpriteSheets[0] == null:
 			push_warning("There are LPCSpriteSheets that are <empty> in the LPCAnimatedSprite2D panel")
-			continue
-		var animatedSprite = AnimatedSprite2D.new()
-		animatedSprite.texture_filter = CanvasItem.TEXTURE_FILTER_NEAREST
-		var spriteFrames = CreateSprites(spriteSheet)
-		animatedSprite.frames = spriteFrames
-		add_child(animatedSprite)
-		if spriteSheet.Name == null || spriteSheet.Name == "":
-			animatedSprite.name = "no_name"
+			return
+		if 1 == SpriteSheets.size():
+			unified_sprite_sheet = SpriteSheets[0]
+			LoadFrames()
 		else:
-			animatedSprite.name = spriteSheet.Name
-		animatedSprite.owner = get_tree().edited_scene_root
-		play(DefaultAnimation)
+			# assuming all spritesheet textures of same dimensions as those of the first.
+			# May be an invalid assumption, as I have no data and can't test - so TBC.
+			var size: Vector2 = SpriteSheets[0].SpriteSheet.get_size()
```

Comment:
Based on this spritesheet [LPC](https://sanderfrenken.github.io/Universal-LPC-Spritesheet-Character-Generator/#?body=Body_color_light&head=Human_male_light&weapon=Scythe_scythe&sex=male) i think that the viewport or should be this size or resize the viewport as necessary (I dont know if it's possible)